internal/app/go-filecoin/plumbing/msg/waiter_test.go

```diff
@@ -51,7 +51,7 @@ func setupTest(t *testing.T) (*hamt.CborIpldStore, *chain.Store, *chain.MessageS
 
 func TestWait(t *testing.T) {
 	tf.UnitTest(t)
-
+	t.Skip("this guy is waiting forever")
```

Comment:
I'm worried that tests like this that exercise parts of the system adjacent to but decoupled from the vm runtime should actually be working after this PR and that we are unnecessarily breaking functionality.  I'd be a lot less worried if there was a description of why tests are broken and after what work they can be expected to be unskipped.

Reply:
Casualty of deleting a lot of the old processor. 
Since the dedup of messages is now done in the VM interpreter I had deleted the "DedupMessages" method from the processor since it was no longer used.

I brought dedupMessages back as a private method on the waiter since nothing else uses that method.

The test has been re-enabled.